src/cryptonote_core/tx_pool.cpp

```diff
@@ -705,6 +705,7 @@ namespace cryptonote
       LOG_PRINT_L2("  added, new block size " << total_size << "/" << max_total_size << ", coinbase " << print_money(best_coinbase));
     }
 
+    expected_reward = best_coinbase;
```

Comment:
The actual coinbase will be coinbase, not best_coinbase, due to the break. best_coinbase is "best so far" to know when to break.